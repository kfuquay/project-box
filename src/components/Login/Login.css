.form-container {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 70vh;
}

.login-form {
    border: 1px solid black;
    max-width: 400px;
    background-color: #444;
    display: flex;
    flex-wrap: wrap;
    flex-direction: column;
    padding: 20px;
}

.form-section {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
}

.form-section > label + * {
    flex: 1 0 120px;
    margin: 8px 0;
}

.form-section > label {
    flex: 1 0 120px;
    max-width: 220px;
    color: whitesmoke;
}

button {
    margin-left: auto;
    padding: 8px 16px;
    border: none;
    background: whitesmoke;
    color: #333;
    text-transform: uppercase;
    letter-spacing: .09em;
    border-radius: 2px;
}

button:hover {
    cursor: pointer;
}