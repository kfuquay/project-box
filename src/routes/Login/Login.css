body {
    background: cadetblue;
}

.form-container {
    width: 80%;
    max-width: 1200px;
    margin: 0 auto;
    padding: 50px 0;
}

.login-form {
    padding: 0;
    max-width: 800px;
    margin: 0 auto;
}

.form-section {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
}

.form-section:not(:last-child) {
    margin-bottom: 20px;
}

.form-section > label {
    padding: 8px;
    font-weight: 300;
    letter-spacing: .09em;
    text-transform: uppercase;
    flex: 1 0 120px;
    max-width: 220px;
}

.form-section > label + * {
    flex: 1 0 220px;
}

.form-section input {
    padding: 15px;
    border: none;
}

.form-section button {
    margin-left: auto;
    padding: 8px 16px;
    border: none;
    background: #333;
    color: #f2f2f2;
    text-transform: uppercase;
    letter-spacing: .09em;
    border-radius: 2px;
}