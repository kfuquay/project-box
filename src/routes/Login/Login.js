import React, { Component } from "react";
import LoginForm from "../../components/LoginForm/LoginForm";

class Login extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {},
    },
  };

 

  render() {
    return (
      <div className="main-container">
        <LoginForm />
      </div>
    );
  }
}
export default Login;
