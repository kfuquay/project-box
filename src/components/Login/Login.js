import React from "react";
import "./Login.css";

function Login() {
  return (
    <div className="form-container">
      <form className="login-form">
        <div className="form-section">
          <label for="username">Username</label>
          <input
            type="text"
            name="username"
            placeholder="knitpearl77"
            required
          />
        </div>
        <div className="form-section">
          <label for="password">Password</label>
          <input type="password" name="password" />
        </div>
        <div className="form-section">
          <button type="submit">Submit</button>
        </div>
      </form>
    </div>
  );
}
export default Login;
