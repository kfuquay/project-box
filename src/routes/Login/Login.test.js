import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Login from './Login'

describe(`Login component`, () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<Login />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
