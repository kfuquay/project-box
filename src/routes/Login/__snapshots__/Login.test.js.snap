// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders without crashing 1`] = `
<div
  className="main-container"
>
  <LoginForm
    onLoginSuccess={[Function]}
  />
</div>
`;
